modules/programs/boxxy.nix

```diff
@@ -0,0 +1,126 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.boxxy;
+
+  boxxyRulesOpts = types.submodule {
+    options = {
```

Comment:
We should add a `freeformType`, see [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#part-1-1)

Reply:
Am on it, just going to run a nix test beforehand to make sure the change doesn't break anything.

Reply:
[This commit](https://github.com/nix-community/home-manager/commit/a1782d051c26e7d833bb1a91ab7dc1e302fa826d) solves this.